dev/generate_bom.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/python
+#
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+import sys
+import yaml
+
+from annotate_source import Annotator
+from spinnaker.run import run_quick
+
+SERVICES = 'services'
+VERSION = 'version'
+
+class BomGenerator(Annotator):
+  """Provides facilities for generating the Bill of Materials file for the
+  Spinnaker product release.
+  """
+
+  COMPONENTS = [
+    'clouddriver',
+    'deck',
+    'echo',
+    'front50',
+    'gate',
+    'igor',
+    'orca',
+    'rosco',
+    'fiat',
+  ]
+
+  def __init__(self, options):
+    self.__base_dir = options.base_dir
+    self.__component_versions = {}
+    super(BomGenerator, self).__init__(options)
+
+  @classmethod
+  def init_argument_parser(cls, parser):
+    """Initialize command-line arguments."""
+    parser.add_argument('--base_dir', default='.',
+                        help="Base directory containing the component's git repositories as subdirectories.")
+    super(BomGenerator, cls).init_argument_parser(parser)
+
+  def write_bom(self):
+    output_yaml = {SERVICES: {}}
```

Comment:
I think the bom should contain some kind of traceability to who/how it was generated, especially if there are different sources of releases (official one vs customer generated)